### PR TITLE
Proposal for multiple resultset support

### PR DESCRIPTION
**Pull request by pmlopes**
see vert-x3/vertx-jdbc-client#80

### CODE REVIEW

**Comment by pmlopes** [comment]
The same concept (multiple resultset must be added to rowstream)

**Comment by pmlopes** [comment]
Fixes vert-x3/issues#145

File: src/main/java/io/vertx/ext/sql/ResultSet.java [hunk 1]
```diff
@@ -22,6 +22,8 @@
   private List<JsonArray> results;
   private List<JsonObject> rows;
   private JsonArray output;
+  // chainned result sets
```
Comment (vietj) [inline src/main/java/io/vertx/ext/sql/ResultSet.java]: typo

File: src/main/java/io/vertx/ext/sql/SQLRowStream.java [hunk 2]
```diff
@@ -17,17 +17,29 @@
...
+   *
+   * @param handler
+   */
+  void close(Handler<AsyncResult<Void>> handler);
+
+  /**
+   * Event handler when a resultset is closed. This is useful to request for more results.
+   * @param handler
+   */
+  SQLRowStream resultSetClosed(Handler<Void> handler);
```
Comment (vietj) [inline src/main/java/io/vertx/ext/sql/SQLRowStream.java]: should be named `resultSetClosedHandler`

File: src/main/asciidoc/js/index.adoc [hunk 3]
```diff
@@ -394,6 +394,17 @@ connection.execute(sql, function (execute, execute_err) {
...
 
+=== Multiple ResultSet responses
+
+In some cases your query might return more than one result set, in this case and to preserve the compatibility when
+the returned result set object is converted to pure json, the next result sets are chained to the current result set
+under the property `next`. A simple walk of all result sets can be achieved like this:
+
+[source,js]
+----
+Code not translatable
```
Comment (vietj) [inline src/main/asciidoc/js/index.adoc]: code cannot be translated to other langs, an issue should be opened in the code translator for this (and a contribution welcome)

File: src/main/asciidoc/groovy/index.adoc [hunk 4]
```diff
@@ -394,6 +394,17 @@ connection.execute(sql, { execute ->
...
 
+=== Multiple ResultSet responses
+
+In some cases your query might return more than one result set, in this case and to preserve the compatibility when
+the returned result set object is converted to pure json, the next result sets are chained to the current result set
+under the property `next`. A simple walk of all result sets can be achieved like this:
+
+[source,groovy]
+----
+Code not translatable
```
Comment (tsegismont) [inline src/main/asciidoc/groovy/index.adoc]: Can you check why the code is not translatable?

Comment (tsegismont) [reply src/main/asciidoc/groovy/index.adoc]: Probably due to the do/while

**Review by tsegismont — DISMISSED** [review]
Looks good to me except the code example which is not translatable

**Pull request merged**